### PR TITLE
Feature/reusable wf for infra

### PR DESCRIPTION
**Pull request by amigos-cosmin**

### CODE REVIEW

File: .github/workflows/cd-rpa-infra.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    needs: [ review-terraform-plan ]
+    runs-on: ubuntu-latest
+    name: Run Terraform Apply
+    environment: dev
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: master
+      - name: Create Release
```
Comment (amigos-sian) [inline .github/workflows/cd-rpa-infra.yml:72]: let's remove release, as I don't think it's necessary. deployments are done per an environment and that provides sufficient tracking

File: .github/workflows/cd-rpa-infra.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          path: ${{ inputs.terraform_path }}
+          var_file: "${{ inputs.terraform_path }}/environments/${{ inputs.environment }}-vars.tfvars"
+          backend_config_file: "${{ inputs.terraform_path }}/environments/${{ inputs.environment }}-backend.tfvars"
+          add_github_comment: false
+
+  terraform-apply:
+    needs: [ review-terraform-plan ]
+    runs-on: ubuntu-latest
+    name: Run Terraform Apply
+    environment: dev
```
Comment (amigos-sian) [inline .github/workflows/cd-rpa-infra.yml:66]: please parameterise this `environment: dev`

File: .github/workflows/cd-rpa-infra.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+name: cd-rpa-infra
+
```
Comment (techamigo) [inline .github/workflows/cd-rpa-infra.yml:2]: This is reusable workflow, so should be generic and no mentions of projects like RPA!

File: .github/workflows/cd-rpa-infra.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+permissions:
+  id-token: write # OIDC token generated in the reusable workflow can be accessed from context of dflook/terraform-plan@v1
+  contents: read
+
+env:
+  # This is required by dflook/terraform-plan@v1, so tf modules can be fetched.
+  TERRAFORM_HTTP_CREDENTIALS: ${{secrets.terraform_http_credentials}}
+  
+jobs:
+  review-terraform-plan:
```
Comment (techamigo) [inline .github/workflows/cd-rpa-infra.yml:37]: This job should wait. How do we ensure that pipeline pauses here once plan is generated for some to look at plan and then approve or even disapprove (cancel without applying)

File: .github/workflows/cd-rpa-infra.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+          aws-region: ${{ inputs.aws_region }}
+      - name: "Create Terraform plan for ${{ inputs.environment }} environment"
+        uses: dflook/terraform-plan@v1
+        with:
+          path: ${{ inputs.terraform_path }}
+          var_file: "${{ inputs.terraform_path }}/environments/${{ inputs.environment }}-vars.tfvars"
+          backend_config_file: "${{ inputs.terraform_path }}/environments/${{ inputs.environment }}-backend.tfvars"
+          add_github_comment: false
+
+  terraform-apply:
```
Comment (techamigo) [inline .github/workflows/cd-rpa-infra.yml:60]: Perhaps think of doing this as separate workflows? one for plan and one for apply? Apply one can have link to the plan? Please check how harness does it, as we are just replicateing it

**Pull request merged**